optimum/habana/transformers/models/modeling_all_models.py

```diff
@@ -164,5 +164,7 @@ def all_reduce(self, input):
             dist.inference_all_reduce(input, group=self.mp_group)
 
     def post_all_reduce(self, input):
-        output = input + self.bias if (self.bias is not None) else input
-        return output
+        # inplace addition needed for correct results
```

Comment:
@skavulya very nice catch! 

Does this affect any other tests or models (falcon, llama, gemma, qwen2, qwen2_moe)?

Reply:
I only looked at llama and qwen2. They were not affected because they didn't use a bias. I can test falcon, and qwen2 moe.

Reply:
@skavulya if llama and qwen2 did not use a bias, then the output is wrong isn't it? since based on the original code `output=input + input`

Reply:
For llama and qwen2, bias was None so the addition is skipped.